### DISCUSSION TITLE
Can Edit in Demo but not in my project

### DISCUSSION THREAD

**Discussion by boduzapho** [TokisanGames/Terrain3D#363] [Questions] [answered]
So I got it to load, thanks to help here BUT when I use it in my project none of the tools work. When I load the demo project they work fine. Also why is the default texture such a bright white color. I have some color blindness so it is almost impossible to see the region boxes because of this. Is there some switch or setting I need to enable to use the tools, and yes I added an area. See screen shot. I did save the res file and the two other tres files as well. Its weird I cant paint or manipulate the mesh and in the tutorial videos it just seems to work once you save the files on a new terrain4d node..

<img width="952" alt="image" src="https://github.com/TokisanGames/Terrain3D/assets/42855504/6f057bbf-410d-4eec-871b-18f571febc3e">

**Comment by boduzapho** [comment]
Also when I click that warning yellow triangle next to the node nothing comes up it is clickable but no UI info box at all

**Comment by boduzapho** [comment]
So I rel;oaded the project to get rid of the warning Icon on the node and it went away BUT the arrays went from 1 to 12
<img width="959" alt="image" src="https://github.com/TokisanGames/Terrain3D/assets/42855504/6ad0e686-614d-472e-a5ef-fe69cb3aa73d">

**Comment by TokisanGames** [comment, marked as answer]
I told you in the other ticket that Terrain3D does not work with the compatibility renderer. You can read more about it in the mobile & web document. Change to Forward+ or Mobile.

The default texture is not bright white. Your material is broken because the render mode is incompatible. 

The warning icon is present when you have the storage resource saved as text or in the scene. Clicking it reports this information. Once you save as .res it goes away. You might have fixed the problem, then clicked the warning before Godot updated the inspector.

The 12 means you probably had the region brush selected and clicked the terrain a bunch of times, thus creating 12 regions for storage to save. Get out of Compatibility mode, then delete all of the extra regions you don't want.

**Reply by boduzapho** [reply]
No I did not click it 12 times, Look man Im just telling you what happened. Dont take reports personally.

**Reply by TokisanGames** [reply]
?? I'm not taking it personally. I just explained the situation and how to fix it. Your system says there are 12 regions. That only happens by clicking the mouse over 12 areas on the terrain with the add region tool. It could have happened in any combination of 12 clicks, or one click and drag. Or you tried to sculpt the terrain across 12 areas, with the add region option enabled (default). Or they could have gotten there had you imported the equivalent of 3k x 4k sized heightmap. The system does not add data for you automatically, only upon user input.

**Comment by boduzapho** [comment]
After wasting a day on answers that should have been articulated much more clearly, examining the reversed engineered logic and structure. I have decided that this plugin is NOT something people should use. Its memory handling is something I would expect from (at BEST) a junior programmer. I give it an F- for efficiency and base logic. My report will be announced on my actual account that does report and reflect on Godot and 2rd party plugins. Sorry kid your just an amateur.
